Tryouts/Visuals/WindowsOnly/DwmUtilsForAvalonia/Dwm/DwmVisualProxy.cs

```diff
@@ -0,0 +1,374 @@
+///
+/// Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License").
+/// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+/// See the NOTICE file distributed with this work for additional information regarding copyright ownership.
+/// Unless required by applicable law or agreed to in writing, software distributed under the License
+/// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and limitations under the License.
+/// 
+/// ********************************************************************************************************
+
+using System;
+using System.Threading;
+using MorganStanley.ComposeUI.Tryouts.Core.Abstractions;
+using Avalonia.Controls;
+using Avalonia;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using Avalonia.VisualTree;
+using Subscriptions;
+using System.Linq;
+using MorganStanley.ComposeUI.Tryouts.Visuals.WindowsOnly.Win32Calls;
+using System.Threading.Tasks;
+
+namespace MorganStanley.ComposeUI.Tryouts.Visuals.WindowsOnly.VisualUtils.Dwm
+{
+    public sealed class DwmVisualProxy : Border
+    {
+        private static float dpiX;
+        private static float dpiY;
+        private IntPtr visualHandle;
+        private IntPtr thumbnailPtr;
+        private Window parentWindow = null;
+        private bool suspended = false;
+
+        static DwmVisualProxy()
+        {
+            using (var graphics = System.Drawing.Graphics.FromHwnd(IntPtr.Zero))
+            {
+                dpiX = graphics.DpiX;
+                dpiY = graphics.DpiY;
+            }
+        }
+
+        private readonly ISubscriptionClient _subscriptionClient;
+
+        IDisposable _boundsSubscription;
+        IDisposable _isVisibleSubscription;
+        public DwmVisualProxy(ISubscriptionClient subscriptionClient, long windowHandle)
+        {
+            _subscriptionClient = subscriptionClient;
+
+            this.Focusable = true;
+
+            this.visualHandle = (IntPtr)windowHandle;
+
+            this.Background = global::Avalonia.Media.Brushes.Transparent;
+            this.AttachedToVisualTree += async (sender, args) =>
+            {
+                await UpdateParentWindow(true);
+            };
+
+            _boundsSubscription = this.GetObservable(BoundsProperty).Subscribe(_ => OnSizeChanged());
+            _isVisibleSubscription = this.GetObservable(IsVisibleProperty).Subscribe(_ => UpdateThumbnailVisibility());
+
+            // WinApi.SendMessage(visualHandle, 0x009B, IntPtr.Zero, IntPtr.Zero);
+
+            RegisterMouseEvents();
+            RegisterKeyboardEvents();
+        }
+
+        private async void OnSizeChanged()
+        {
+            _subscriptionClient.Publish(Topic.WindowSize, new WindowCoordMessage { WindowId = (long)visualHandle, X = (int)Bounds.Width, Y = (int)Bounds.Height});
```

Comment:
Do you need to 'await' this operation?

Reply:
No need to await on this operation, since it does not return anything - just publishes a message.

Reply:
You can use a discard to be explicit about the intention that this is just fire-and-forget, no need to await the Task which was returned.
```
_ = _subscriptionClient.Publish(...)
```